### ISSUE TITLE
xs-vat-worker limited to one at a time? (hard-coded fd 3, 4)

### ISSUE THREAD

**Issue by dckc** [Agoric/agoric-sdk#1616]
context: https://github.com/Agoric/agoric-sdk/pull/1407#discussion_r476189291

cc @warner

**Comment by warner** [comment]
yeah we should add command-line arguments to specify the fds that the child should use, the parent will allocate a pair for each worker process it launches and tell the child what it picked

**Comment by dckc** [comment]
I wonder... Many children can each have fd 3 and 4, no?

**Comment by warner** [comment]
Yeah, but the parent cannot. But, now you're making me wonder how this really works, and whether the FD numbers have to be the same in both child and parent (I know it's doing fork/exec, but the parent process could `close`/`fdup` to reassign the numbers before it `exec`s the child). I wrote code to do this in Twisted 20 years ago, but for the life of me I don't remember what I did.

We'll have to check the Node.js `spawn` docs to see if this is strictly necessary.
